### PR TITLE
Remove scheduler state manager adaptor requirement for submission dry-run

### LINKED ISSUE #1692: Accelerate dry-run submission
Dry-run submission does not need to use state manager. However currently we still initialize state manager before dry-run (see [here](https://github.com/twitter/heron/blob/master/heron/scheduler-core/src/java/com/twitter/heron/scheduler/SubmitterMain.java#L414)). This could increase dry-run process for about five seconds. 

We should definitely optimize it out.

### PR DESCRIPTION
**Pull request by objmagic**
Dry-run submission does not need to use adaptor to scheduler state manager. With this PR, state manager initialization will not happen, which reduces dry-run of submission to Aurora from 10 seconds to 1.7 seconds.

resolves #1692

### CODE REVIEW

**Comment by objmagic** [comment]
Also addressed [some comments](https://github.com/twitter/heron/pull/1629#discussion_r99461641) from @mycFelix in https://github.com/twitter/heron/pull/1697/commits/1e34ccc490bf6fd88a5169e744e699f34f71be0d

File: heron/scheduler-core/src/java/com/twitter/heron/scheduler/dryrun/FormatterUtils.java [hunk 1]
```diff
@@ -381,6 +381,9 @@ public String createTable() {
     }
   }
 
+  private static final List<String> titleNames = Arrays.asList(
```
Comment (billonahill) [inline heron/scheduler-core/src/java/com/twitter/heron/scheduler/dryrun/FormatterUtils.java]: psf should be TITLE_NAMES

File: heron/scheduler-core/src/java/com/twitter/heron/scheduler/SubmitterMain.java [hunk 2]
```diff
@@ -410,20 +410,19 @@ public void submitTopology() throws TopologySubmissionException {
...
+      Config primaryRuntime = Config.newBuilder()
+          .putAll(LauncherUtils.getInstance().createPrimaryRuntime(topology)).build();
 
       // Bypass validation and upload if in dry-run mode
       if (Context.dryRun(config)) {
-        callLauncherRunner(runtime);
+        callLauncherRunner(primaryRuntime);
       } else {
+        // initialize the state manager
+        statemgr.initialize(config);
```
Comment (huijunw) [inline heron/scheduler-core/src/java/com/twitter/heron/scheduler/SubmitterMain.java]: +1

Comment (objmagic) [inline heron/scheduler-core/src/java/com/twitter/heron/scheduler/SubmitterMain.java]: what do you mean?

Comment (huijunw) [inline heron/scheduler-core/src/java/com/twitter/heron/scheduler/SubmitterMain.java]: looks good to me. it is expected as the PR description.

Comment (objmagic) [inline heron/scheduler-core/src/java/com/twitter/heron/scheduler/SubmitterMain.java]: you can add a reply next time instead of making an inline comment.

**Comment by objmagic** [comment]
build is fine on CI, unit tests passed locally. Website build breaks the CI because of some package issues, which will be solved in #1703. 

merging now.

**Pull request merged**